Sources/SymDiff/source/LiveVariablesAnalysis.cs

```diff
@@ -75,31 +75,29 @@ public void ComputeLiveVariables(Implementation impl, HashSet<GlobalVariable> gl
         if (cmds[i] is CallCmd)
         {
           Procedure /*!*/
-            proc = cce.NonNull(cce.NonNull((CallCmd /*!*/) cmds[i]).Proc);
+            proc = ((CallCmd /*!*/) cmds[i]).Proc;
           if (InterProcGenKill.HasSummary(proc.Name)) // TODO: this is always false! InterProcGenKill.ComputeLiveVars is never called.
           {
             liveVarsAfter =
-              InterProcGenKill.PropagateLiveVarsAcrossCall(options, cce.NonNull((CallCmd /*!*/) cmds[i]), liveVarsAfter);
+              InterProcGenKill.PropagateLiveVarsAcrossCall(options, (CallCmd /*!*/) cmds[i], liveVarsAfter);
             continue;
           }
         }
 
         Propagate(cmds[i], liveVarsAfter);
       }
 
-      block.liveVarsBefore = liveVarsAfter;
+      block.LiveVarsBefore = liveVarsAfter;
     }
   }
 
   // perform in place update of liveSet
   public void Propagate(Cmd cmd, HashSet<Variable /*!*/> /*!*/ liveSet)
   {
     Contract.Requires(cmd != null);
-    Contract.Requires(cce.NonNullElements(liveSet));
-    if (cmd is AssignCmd)
-    {
+    if (cmd is AssignCmd) {
       AssignCmd /*!*/
-        assignCmd = (AssignCmd) cce.NonNull(cmd);
```

Comment:
Do you need a `Contract.Assert(cmd != null)` here?

Reply:
If the `cmd is AssignCmd` check succeeds, `cmd` is guaranteed to be non-null.